src/CoreWCF.NetTcp/tests/CompressionFormatTests.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using CoreWCF.Channels;
+using CoreWCF.Configuration;
+using Helpers;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace CoreWCF.NetTcp.Tests
+{
+    public class CompressionFormatTests
+    {
+        private readonly ITestOutputHelper _output;
+
+        public CompressionFormatTests(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Theory]
+        [MemberData(nameof(GetTestVariations))]
+        public void CompressionFormatTest(
+            MessageVersion messageVersion,
+            System.ServiceModel.Channels.MessageVersion clientMessageVersion,
+            CompressionFormat compressionFormat,
+            System.ServiceModel.Channels.CompressionFormat clientCompressionFormat,
+            TransferMode transferMode,
+            System.ServiceModel.TransferMode clientTransferMode)
+        {
+            string testString = new string('a', 8000);
+            var webHostBuilder = ServiceHelper.CreateWebHostBuilder<Tests.Startup>(_output);
+            webHostBuilder.ConfigureServices(services => services.AddServiceModelServices());
+            webHostBuilder.Configure(app =>
+            {
+                app.UseServiceModel(builder =>
+                {
+                    builder.AddService<Services.TestService>();
+                    builder.AddServiceEndpoint<Services.TestService, ServiceContract.ITestService>(GetServerBinding(messageVersion, compressionFormat, transferMode), "/MessageVersionTest.svc");
+                });
+            });
+            var host = webHostBuilder.Build();
+            using (host)
+            {
+                host.Start();
+                var factory = new System.ServiceModel.ChannelFactory<ClientContract.ITestService>(GetClientBinding(clientMessageVersion, clientCompressionFormat, clientTransferMode),
+                    new System.ServiceModel.EndpointAddress(host.GetNetTcpAddressInUse() + "/MessageVersionTest.svc"));
+                ClientContract.ITestService channel = factory.CreateChannel();
+                var result = channel.EchoString(testString);
+                Assert.Equal(testString, result);
+                ((System.ServiceModel.IClientChannel)channel).Close();
+            }
+        }
+
+        public static IEnumerable<object[]> GetTestVariations()
+        {
+            yield return new object[]
+            {
```

Comment:
You can simplify this by using loops. You can also cast an enum between types as they are backed by an integer. You could do something like this:
```c#
foreach (CompressionFormat compression in Enum.GetValues(typeof(CompressionFormat)))
{
    foreach (TransferMode transferMode in Enum.GetValues(typeof(TransferMode))
    {
        yield return new object[]
        {
                MessageVersion.Soap12WSAddressing10,
                System.ServiceModel.Channels.MessageVersion.CreateVersion(System.ServiceModel.EnvelopeVersion.Soap12, System.ServiceModel.Channels.AddressingVersion.WSAddressing10),
                compression,
                (System.ServiceModel.Channels.CompressionFormat)compression,
                transferMode,
                (System.ServiceModel.TransferMode)transferMode
        };
        yield return new object[]
        {
                MessageVersion.Soap12WSAddressingAugust2004,,
                System.ServiceModel.Channels.MessageVersion.Soap12WSAddressingAugust2004,
                compression,
                (System.ServiceModel.Channels.CompressionFormat)compression,
                transferMode,
                (System.ServiceModel.TransferMode)transferMode
        };
    }
}
```
I haven't tested this code so you might need to cast to int first when casting between the CoreWCF and the SM variations, but I think the above will work. Also GetValues returns an Array which has a signature of returning objects. I think the foreach does an implicit cast but it might need the iterator variable to be type object and then cast where it's used. I believe it should work as written, but there's a chance you might need to make these changes.

Reply:
Thanks for the suggestion- I've added it in my latest commit. Looks great, worked perfectly, and increased the test coverage from 8 to 24 cases.